1 - DataTypes/Code/WeirdCalculator.cs

```diff
@@ -40,28 +42,68 @@ public static decimal ComplexDecimalAdd(float firstValue, double secondValue, de
 			 * If you are new to exceptions, google "C# Exception Handling" - litterally half our jobs is googling :P
 			 */
 
-			throw new NotImplementedException();
+            decimal first;
+            decimal second;
+            
+            try
+            {
+                first = (decimal)firstValue;
+            }
+            catch
+            {
+                return 0;
+            }
+            try
+            {
+                second = (decimal)secondValue;
+            }
+            catch
+            {
+                return 0;
+            }
+
+            return first + second + thirdValue;
+            
 		}
 
 		public static decimal ComplexAdd(char firstValue, string secondValue)
 		{
-			/*
+            /*
 			 * You will notice the TDD tests around this just "assume" it will work. Do not worry about handling exceptions or errors, assume input is valid.
 			 */
-
-			throw new NotImplementedException();
+            return decimal.Parse(firstValue.ToString())+decimal.Parse(secondValue);
```

Comment:
Perfect!

An alternative for turning a char into a numeric value is using the `Char` static class for it has a function `GetNumericValue`.